### PR TITLE
Issue #6236: Fix image browser in modal

### LINKED ISSUE #6236: Image browser attached to field doesnt work if field is in a modal
## Description of the bug

If you implement a `managed_file` field and add the `#browser_view` key to the array, the button to browse existing files appears, and when clicked, launches the image browser in a modal. This works fine on forms like the node page, but what if your field is already in a modal? Well, the image browser modal pops out, and when you click the insert button closes the original modal too. 

## Steps To Reproduce

Simplest is to hack core: go to `core/layout/includes/block.hero.inc` and add `'#browser_view' => 'image_library',` to the `$form['image']` array in the `form()` function. Then start add a Hero block to any layout, and you get this:

Screenshot 2023-09-30 140307

## Actual behavior

But if you try to select existing file, after clicking `insert` it closes the Hero block config modal. The browser works fine if you add the `#browser_view` key to a field that's not in a modal. (Feel free to hack another core or contrib `managed_file` field to see that it works fine otherwise).

## Expected behavior

Clicking `insert` should return you to your previous modal, with the selected image active in the image field.

### PR DESCRIPTION
**Pull request by docwilmot**
Fixes https://github.com/backdrop/backdrop-issues/issues/6236

### CODE REVIEW

File: core/modules/file/file.module [hunk 1]
```diff
@@ -1744,7 +1745,7 @@ function file_managed_file_browser_open($form, $form_state) {
 
   $commands = array();
   $commands[] = ajax_command_settings($settings, TRUE);
-  $commands[] = ajax_command_open_modal_dialog($title, $html, $options);
+  $commands[] = ajax_command_open_dialog('#backdrop-browser-modal', $title, $html, $options);;
```
Comment (laryn) [inline core/modules/file/file.module]: There's an extra `;` here.

Comment (docwilmot) [reply core/modules/file/file.module]: Fixed

File: core/modules/file/file.module [hunk 2]
```diff
@@ -1744,7 +1745,7 @@ function file_managed_file_browser_open($form, $form_state) {
 
   $commands = array();
   $commands[] = ajax_command_settings($settings, TRUE);
-  $commands[] = ajax_command_open_modal_dialog($title, $html, $options);
+  $commands[] = ajax_command_open_dialog('#backdrop-browser-modal', $title, $html, $options);
```
Comment (quicksketch) [inline core/modules/file/file.module]: The other selectors in this function are labeled `file-browser-*`. Could we change this to `#file-browser-modal`?

Comment (docwilmot) [reply core/modules/file/file.module]: Done

**Comment by quicksketch** [comment]
Thanks @docwilmot! This works great. Merged into 1.x and 1.26.x.

**Pull request merged**